File: clients/lumify-sdk/src/resources/sports.ts
import type { LumifyClient } from "../client.js";
import type { SeasonsListResponse, SportsListResponse } from "../generated/models.js";

export interface ListSportsParams {
  /** Only return currently-active sports. */
  activeOnly?: boolean;
}

export class SportsResource {
  constructor(private readonly client: LumifyClient) {}

  /** GET /v1/sports — every sport Lumify covers (mlb, nfl, nba, nhl, tennis, soccer, ncaaf, ncaab, ...). */
  list(params: ListSportsParams = {}): Promise<SportsListResponse> {
    return this.client.get<SportsListResponse>("/v1/sports", {
      query: { active_only: params.activeOnly },
    });
  }
}

export interface ListSeasonsParams {
  /** Sport slug filter (e.g. "nfl"). */
  sport?: string;
  /** Only return each sport/league's current season. */
  currentOnly?: boolean;
}

export class SeasonsResource {
  constructor(private readonly client: LumifyClient) {}

  /** GET /v1/seasons */
  list(params: ListSeasonsParams = {}): Promise<SeasonsListResponse> {
    return this.client.get<SeasonsListResponse>("/v1/seasons", {
      query: { sport: params.sport, current_only: params.currentOnly },
    });
  }
}

File: clients/lumify-sdk-python/lumify/resources/sports.py
from __future__ import annotations

from typing import Optional

from .._transport import LumifyClient
from ..models import SeasonsListResponse, SportsListResponse


class SportsResource:
    def __init__(self, client: LumifyClient) -> None:
        self._client = client

    def list(self, *, active_only: Optional[bool] = None) -> SportsListResponse:
        """GET /v1/sports — every sport Lumify covers (mlb, nfl, nba, nhl,
        tennis, soccer, ncaaf, ncaab, ...)."""
        return self._client.get("/v1/sports", query={"active_only": active_only})


class SeasonsResource:
    def __init__(self, client: LumifyClient) -> None:
        self._client = client

    def list(
        self,
        *,
        sport: Optional[str] = None,
        current_only: Optional[bool] = None,
    ) -> SeasonsListResponse:
        """GET /v1/seasons — seasons across sports/leagues."""
        return self._client.get(
            "/v1/seasons", query={"sport": sport, "current_only": current_only}
        )
